### PR TITLE
Fixes #266. Update CODEOWNERS for Surface

### LINKED ISSUE #266: Update CodeOwners for Surface components
@mathomp4  I forwarded the spreadsheet of assignments separately.
(cc'ing @gmao-rreichle )

### PR DESCRIPTION
**Pull request by mathomp4**
I think this encompasses most of the changes @gmao-rreichle wanted in #266. Take a look and see. Note that this PR is against `master` since that's the "overall" `CODEOWNERS` and the one that Github seems to see best. If it's acceptable, I'll then make a PR to put it into `develop`.

### CODE REVIEW

**Comment by smahanam** [comment]
Sorry 1 I replied in a wrong discussion.

I am not sure I understand everything that is going on. I did not know whether I was even a code owner - Randy was the lead scientist, I have been maintaining land science developments.
If this ownership change means transferring the ongoing work to the @GEOS-ESM/land-team, both GEOScatchCN_GridComp and GEOSroute_GridComp are not ready.
GEOScatchCN_GridComp: I created and maintained but lost the track during CVS-git transfer so I need to verify few things and make sure we transferred from CVS to git correctly. And there is another pending issue, I can prepare this transfer soon.
GEOSroute sure needs me to finish it. We have a ticket with the SI-team, once that issue is resolved only I can finish GEOSroute_GridComp

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -9,5 +9,31 @@
...
+/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/            @GEOS-ESM/land-team
+
+# The surface preproc team owns these
+/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/mk_restarts/ @GEOS-ESM/surface-preproc-team
+/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/                                    @GEOS-ESM/surface-preproc-team
+
+# But the OAsim file is ocean
+/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/OASIMalbedoMod.f @GEOS-ESM/ocean-team
+
+# Landice Team owns Landice
```
Comment (gmao-rreichle) [inline .github/CODEOWNERS:25]: Would it be more logical if this was moved up a bit, so it appears before the "surface-preproc-team" entries?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -9,5 +9,31 @@
...
+/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSlandice_GridComp/ @GEOS-ESM/landice-team
+
+# Some files are shared...
+/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/StieglitzSnow.F90 @GEOS-ESM/land-team @GEOS-ESM/landice-team
+/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/SurfParams.F90    @GEOS-ESM/land-team @GEOS-ESM/landice-team
+
+# ...these are duped, so don't do a full path
+cleanup_code.perl     @GEOS-ESM/land-team @GEOS-ESM/surface-preproc-team
```
Comment (gmao-rreichle) [inline .github/CODEOWNERS:33]: I would leave the "duplicates" out of the ownership list for now.
The duplicates should be cleaned up in a future pull request.
We can revisit ownership at that time if needed.

File: .github/CODEOWNERS (hunk 1 above)
Comment (mathomp4) [reply .github/CODEOWNERS:25]: If you like. It's equivalent, but I can do that.

File: .github/CODEOWNERS (hunk 2 above)
Comment (mathomp4) [reply .github/CODEOWNERS:33]: Okay. That means @GEOS-ESM/land-team will own them, I think.

File: .github/CODEOWNERS (hunk 1 above)
Comment (mathomp4) [reply .github/CODEOWNERS:25]: Done. Please re-inspect.

File: .github/CODEOWNERS (hunk 2 above)
Comment (mathomp4) [reply .github/CODEOWNERS]: Done.

**Pull request merged**